driver-core/src/main/com/mongodb/internal/connection/ConcurrentPool.java

```diff
@@ -384,8 +384,9 @@ boolean acquirePermitFair(final long timeout, final TimeUnit unit) throws MongoI
                 throw new MongoInterruptedException(null, e);
             }
             try {
-                while (permits == 0) {
-                    throwIfClosedOrPaused();
+                while (permits == 0
+                        // the absence of short-circuiting is of importance
+                        & !throwIfClosedOrPaused()) {
```

Comment:
This isn't fixing a bug, right? If not, consider holding off on it until the next PR, where we're modifying this loop anyway.

Reply:
This change just applies the same style as in `DefaultConnectionPool`. This change has at least some relevance to the  current PR, but will be completely irrelevant und weird in the next PR. Why do you think it is better to do it there?

Reply:
We discussed this over Zoom, and decided that the style change in `ConcurrentPool` will be committed separately.

Reply:
I force-pushed changes, it seemed appropriate here. Done.